src/proj.cpp

```diff
@@ -228,12 +245,29 @@ Rcpp::NumericMatrix CPL_proj_direct(Rcpp::CharacterVector from_to, Rcpp::Numeric
 
 //	for (int i = 0; i < pts.nrow(); i++)
 //  		 Rcout << xx[i] << " " << yy[i] << std::endl;
-
-	if (pj_transform(fromPJ, toPJ, pts.nrow(), 0, xx.data(), yy.data(), NULL) != 0) {
-		pj_free(fromPJ); pj_free(toPJ); // #nocov start
-		Rcout << "error in pj_transform: " << pj_strerrno(*pj_get_errno_ref()) << std::endl;
-		stop("error"); // #nocov end
-	}
+        if (keep[0] == 1) {
+            // use proj_trans() on individual points, making unprojectable points be NA
+            // FIXME: not tested, since author has no access to the old proj API.
+            double thisx, thisy;
+            for (int i = 0; i < pts.nrow(); i++) {
+                thisx = xx[i];
+                thisy = yy[i];
+                if (pj_transform(fromPJ, toPJ, 1, 0, &thisx, &thisy, NULL) != 0) {
+                    xx[i] = NA_REAL;
```

Comment:
According to the documentation, +Inf or -Inf values should be returned, and I guess this is what the PROJ6 code does.

Reply:
Yes, you are correct. I pushed an update to my fork, changing the assigment from `NA_REAL` to `R_PosInf`.  (I do not know whether to be doing new pull requests for these changes, or posting patches in these comments.  I apologize of my answering in this way is making more work for you, in addition to that occasioned by my sloppy coding.)

Reply:
Did/does `rgdal` also return `Inf` values for unprojectable points?

Reply:
Yes, e.g. as below. I don't know about whether it ever gives `-Inf`, and so in my update to `src/proj.cpp` I made unprojectable points turn into `Inf`.
```R
> rgdal::project(pts, "+proj=lcc +lat_0=30 +lat_1=60")
     [,1]     [,2]
[1,]    0 -4329324
[2,]  Inf      Inf
[3,]    0 -4329324
```